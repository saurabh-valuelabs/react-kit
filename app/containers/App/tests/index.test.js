/**
 *
 * Tests for App
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { App } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';
import configureStore from '../../../configureStore';

import nav from '../../../config/_nav';
import routes from '../../../config/_routes';

jest.mock('react-router-dom', () => ({
  Link: 'a',
  Route: ({ children }) =>
    typeof children === 'function' ? children({ match: false }) : 'div',
  Switch: ({ children }) =>
    typeof children === 'function' ? children({ match: false }) : 'div',
}));

describe('<App />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <App
            dispatch={dispatch}
            dGetAppDetail={jest.fn()}
            isLoading={false}
            hasError={false}
            error=""
            meta={{}}
            headerBlock={{
              logoSrc: '',
              logoAltText: 'NepalFranchise',
              headerNavList: nav,
            }}
            passedRoutes={routes}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to have additional unit tests specified', () => {
    expect(true).toEqual(true);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <App
            dGetAppDetail={jest.fn()}
            isLoading={false}
            hasError={false}
            error=""
            meta={{}}
            headerBlock={{
              logoSrc: '',
              logoAltText: 'NepalFranchise',
              headerNavList: nav,
            }}
            passedRoutes={routes}
          />
        </IntlProvider>
      </Provider>,
    );

    expect(firstChild).toMatchSnapshot();
  });
});
