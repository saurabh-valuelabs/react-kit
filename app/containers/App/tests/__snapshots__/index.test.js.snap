// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render and match the snapshot 1`] = `
<div
  class="container"
>
  <header>
    <nav
      class="row px-0 navbar navbar-expand-lg navbar"
    >
      <div
        class="col-6 col-lg-auto"
      >
        <a
          class="navbar-brand"
          to="/"
        >
          <img
            alt="NepalFranchise"
            src=""
          />
        </a>
      </div>
      <div
        class="lg-hide col-12"
      >
        <button
          aria-hidden="true"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
      </div>
      <div
        class="col"
      >
        <div
          class="collapse navbar-collapse"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                title="Dashboard"
                to="/"
              >
                Dashboard
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                title="Funds Listing"
                to="/fund-listing"
              >
                Funds Listing
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                title="My Profile"
                to="/my-profile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <hr />
  </header>
  <main>
    div
  </main>
  <footer>
    <hr />
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="col-auto"
      >
        <button
          aria-label="Back To Top"
          class="btn btn-link"
          type="button"
        >
          <span>
            Back To Top
          </span>
        </button>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-12 col-md-4 col-lg-2"
      >
        <h4>
          About Us
        </h4>
        <ul
          class="nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="px-0 nav-link"
              to="/privacy-policy"
            >
              Privacy Policy
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="px-0 nav-link"
              to="/list"
            >
              Term and Condition
            </a>
          </li>
        </ul>
      </div>
    </div>
  </footer>
</div>
`;
